src/Lightning/NoiseProtocol/KeyGenerator.cs

```diff
@@ -0,0 +1,13 @@
+using System;
+using System.Linq;
+using NBitcoin;
+
+namespace NoiseProtocol
+{
+   public class KeyGenerator : IKeyGenerator
+   {
+      public byte[] GenerateKey() => new Key().ToBytes();
```

Comment:
Do we want to relay on the random key generator inside the Key class?

Reply:
We already use NBitcoin for the EC multiply operation so no real impact here in calling it as well.